src/mscorlib/src/System/Collections/Generic/ArraySortHelper.cs

```diff
@@ -148,6 +148,42 @@ public int BinarySearch(T[] array, int index, int length, T value, IComparer<T>
 
         #endregion
 
+        internal static void Sort(T[] keys, int index, int length, Comparison<T> comparer)
+        {
+            Contract.Assert(keys != null, "Check the arguments in the caller!");
+            Contract.Assert(index >= 0 && length >= 0 && (keys.Length - index >= length), "Check the arguments in the caller!");
+            Contract.Assert(comparer != null, "Check the arguments in the caller!");
+
+            // Add a try block here to detect bogus comparisons
+            try
+            {
+#if FEATURE_CORECLR
+                // Since QuickSort and IntrospectiveSort produce different sorting sequence for equal keys the upgrade 
+                // to IntrospectiveSort was quirked. However since the phone builds always shipped with the new sort aka 
+                // IntrospectiveSort and we would want to continue using this sort moving forward CoreCLR always uses the new sort.
+
+                IntrospectiveSort(keys, index, length, comparer);
+#else
+                if (BinaryCompatibility.TargetsAtLeast_Desktop_V4_5)
```

Comment:
Yes, it is fine to delete the code under `#if !FEATURE_CORECLR`. https://github.com/dotnet/coreclr/issues/972#issuecomment-257633318